### ISSUE TITLE
normalize_name: undefined method sub for nil:NilClass

### ISSUE THREAD

**Issue by optiz0r** [rodjek/librarian-puppet#333]
Encountered this today with librarian-puppet 2.2.1, librarianp 0.6.3, ruby 2.1.7p400 (sabayon, x86_64)

```
ben@triumph ~/g/puppet:development> librarian-puppet update --verbose herculesteam-augeasproviders
[Librarian] Ruby Version: 2.1.7
[Librarian] Ruby Platform: x86_64-linux
[Librarian] Rubygems Version: 2.4.8
[Librarian] Librarian Version: 0.6.3
[Librarian] Librarian Adapter: puppet
[Librarian] Librarian Adapter Version: 2.2.1
[Librarian] Project: /home/ben/git/puppet
[Librarian] Specfile: Puppetfile
[Librarian] Lockfile: Puppetfile.lock
[Librarian] Git: /usr/bin/git
[Librarian] Git Version: 2.6.0
[Librarian] Git Environment Variables:
[Librarian]   (empty)
[Librarian] Pre-Cached Sources:
[Librarian]   [:forge, "https://forgeapi.puppetlabs.com", {}]
[Librarian]   [:git, "https://github.com/kwilczynski/puppet-functions.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-acl.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppetlabs-firewall.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-jenkins.git", {:ref=>"features/sabayon_support"}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-nrpe.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppetlabs-rabbitmq.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-webapp-config.git", {}]
[Librarian] Post-Cached Sources:
[Librarian]   [:forge, "https://forgeapi.puppetlabs.com", {}]
[Librarian]   [:git, "https://github.com/kwilczynski/puppet-functions.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-acl.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppetlabs-firewall.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-jenkins.git", {:ref=>"features/sabayon_support"}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-nrpe.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppetlabs-rabbitmq.git", {}]
[Librarian]   [:git, "https://github.com/optiz0r/puppet-webapp-config.git", {}]
[Librarian] Resolving herculesteam-augeasproviders (>= 0) <https://forgeapi.puppetlabs.com>
[Librarian]   Checking manifests
[Librarian]   Module herculesteam-augeasproviders found versions: 2.1.3, 2.1.2, 2.1.1, 2.1.0, 2.0.0
[Librarian]     Checking herculesteam-augeasproviders/2.1.3 <https://forgeapi.puppetlabs.com>
/usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/util.rb:51:in `normalize_name': undefined method `sub' for nil:NilClass (NoMethodError)
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/dependency.rb:12:in `initialize'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/source/forge.rb:147:in `new'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/source/forge.rb:147:in `block in fetch_dependencies'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/source/forge.rb:145:in `each'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/source/forge.rb:145:in `map'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/source/forge.rb:145:in `fetch_dependencies'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/manifest.rb:97:in `fetch_dependencies!'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/manifest.rb:89:in `fetched_dependencies'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/manifest.rb:46:in `dependencies'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/manifest.rb:63:in `exclude_dependencies!'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:67:in `block in do_resolve'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:161:in `block (3 levels) in resolving_dependency_map_find_manifests'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:194:in `block in scope_checking_manifest'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:230:in `scope'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:193:in `scope_checking_manifest'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:160:in `block (2 levels) in resolving_dependency_map_find_manifests'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:218:in `block in map_find'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:217:in `each'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:217:in `map_find'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:159:in `block in resolving_dependency_map_find_manifests'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:172:in `block (2 levels) in scope_resolving_dependency'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:186:in `block in scope_checking_manifests'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:230:in `scope'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:185:in `scope_checking_manifests'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:171:in `block in scope_resolving_dependency'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:230:in `scope'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:170:in `scope_resolving_dependency'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:158:in `resolving_dependency_map_find_manifests'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:65:in `do_resolve'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver/implementation.rb:50:in `resolve'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/resolver.rb:23:in `resolve'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/action/update.rb:24:in `run'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/cli.rb:173:in `update!'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/cli.rb:115:in `update'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/lib/librarian/puppet/cli.rb:76:in `update'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /usr/lib64/ruby/gems/2.1.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/cli.rb:26:in `block (2 levels) in bin!'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/cli.rb:31:in `returning_status'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/cli.rb:26:in `block in bin!'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/cli.rb:47:in `with_environment'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarianp-0.6.3/lib/librarian/cli.rb:26:in `bin!'
        from /usr/lib64/ruby/gems/2.1.0/gems/librarian-puppet-2.2.1/bin/librarian-puppet:7:in `<top (required)>'
        from /usr/bin/librarian-puppet:9:in `load'
        from /usr/bin/librarian-puppet:9:in `<main>'
```

With the help of a little puts debugging narrowed this down to `RepoV3.dependencies`:
```ruby
          def dependencies(version)
            array = get_release(version).metadata[:dependencies].map{|d| [d['name'], d['version_requirement']]}
            Hash[*array.flatten(1)]
          end
```

Where the array looks like:
```
{:name=>"puppetlabs/stdlib", :version_requirement=>">=3.2.0 <5.0.0"}
{:name=>"herculesteam/augeasproviders_apache", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_base", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_grub", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_mounttab", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_nagios", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_pam", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_postgresql", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_puppet", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_shellvar", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_ssh", :version_requirement=>">=2.1.0 <3.0.0"}
{:name=>"herculesteam/augeasproviders_sysctl", :version_requirement=>"2.x"}
{:name=>"herculesteam/augeasproviders_syslog", :version_requirement=>"2.x"}
```

A little hackery swapping out the strings for symbols seems to resolve the problem for me:
```ruby
          def dependencies(version)
            array = get_release(version).metadata[:dependencies].map{|d| [d[:name], d[:version_requirement]]}
            Hash[*array.flatten(1)]
          end
```

Not sure what the underlying cause is and whether this is a proper fix, but seems to work here.

**Comment by nvanheuverzwijn** [comment]
This fix works. The file he's talking about is `/var/lib/gems/2.3.0/gems/librarian-puppet-2.1.0/lib/librarian/puppet/source/forge/repo_v3.rb` or something alike depending on your distro.

You simply have to replace `'name'` for `:name` and `'version_requirement'` for `:version_requirement`.

**Comment by optiz0r** [comment]
Fixed by voxpupuli/librarian-puppet@249179cac845acb95fad934ead95fa024ef80e04, present in v2.2.2+
